src/Microsoft.Diagnostics.Monitoring.RestServer/DiagnosticServices.cs

```diff
@@ -114,82 +112,46 @@ private static DumpType MapDumpType(Models.DumpType dumpType)
             }
 
-            // Short-circuit for when running in a Docker container.
-            if (RuntimeInfo.IsInDockerContainer)
-            {
-                try
-                {
-                    IProcessInfo processInfo = GetSingleProcessInfo(
-                        processInfos,
-                        DockerEntrypointProcessFilter);
-
-                    using var timeoutSource = new CancellationTokenSource(DockerEntrypointWaitTimeout);
-
-                    var client = new DiagnosticsClient(processInfo.EndpointInfo.Endpoint);
-                    await client.WaitForConnectionAsync(timeoutSource.Token);
-
-                    return processInfo;
-                }
-                catch
-                {
-                    // Process ID 1 doesn't exist, didn't advertise in connect mode, or is not a .NET process.
-                }
-            }
-
-            return GetSingleProcessInfo(
-                processInfos,
-                processKey: null);
+            return processes;
         }
 
-        private IProcessInfo GetSingleProcessInfo(IEnumerable<IProcessInfo> processInfos, ProcessKey? processKey)
+        public Task<IProcessInfo> GetProcessAsync(ProcessKey? processKey, CancellationToken token)
         {
+            ProcessFilterOptions filterOptions = _defaultProcessOptions;
             if (processKey.HasValue)
             {
-                if (processKey.Value.RuntimeInstanceCookie.HasValue)
-                {
-                    Guid cookie = processKey.Value.RuntimeInstanceCookie.Value;
-                    processInfos = processInfos.Where(info => info.EndpointInfo.RuntimeInstanceCookie == cookie);
-                }
+                filterOptions = new ProcessFilterOptions();
+                filterOptions.ProcessFilters.Add(TransformKey(processKey.Value));
+            }
 
-                if (processKey.Value.ProcessId.HasValue)
-                {
-                    int pid = processKey.Value.ProcessId.Value;
-                    processInfos = processInfos.Where(info => info.EndpointInfo.ProcessId == pid);
-                }
+            return GetProcessAsync(filterOptions, token);
+        }
 
-                string processName = processKey.Value.ProcessName;
-                if (!string.IsNullOrEmpty(processName))
-                {
-                    processInfos = processInfos.Where(info => string.Equals(info.ProcessName, processName, StringComparison.OrdinalIgnoreCase));
-                }
+        private async Task<IProcessInfo> GetProcessAsync(ProcessFilterOptions processFilterConfig, CancellationToken token)
+        {
+            //Short circuit when we are missing default process config
+            if (!processFilterConfig.ProcessFilters.Any())
+            {
+                throw new InvalidOperationException("No default process configuration has been set.");
```

Comment:
Is this the message in the HTTP response when the default process is not configured?

Reply:
Yep